### PR TITLE
fix set_cipherlist edge case

### PR DESCRIPTION
**Pull request by miyazakh**
# Description

CC-RX malloc returns NULL when its size is zero. This causes wolfSSL_CTX_set_cipher_list failure in edge case.

./configure --enable-opensslextra --enable-smallstack

# Testing
Run unit test case
Confirm the fix by CC-RX

# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by miyazakh** [comment]
retest this please

File: src/ssl.c [hunk 1]
```diff
@@ -12708,8 +12708,19 @@ static int wolfSSL_parse_cipher_list(WOLFSSL_CTX* ctx, Suites* suites,
 
 #ifdef WOLFSSL_SMALL_STACK
     suitesCpy = (byte*)XMALLOC(suites->suiteSz, NULL, DYNAMIC_TYPE_TMP_BUFFER);
-    if (suitesCpy == NULL)
-        return WOLFSSL_FAILURE;
+    if (suitesCpy == NULL) {
```
Comment (cconlon) [inline src/ssl.c:12711]: This seems a little odd to me, that if `suites->suiteSz` is 0 that we would allocate `suitesCpy` to a size of `1`.

Is this the correct way to handle this, or should we just be skipping the `XMALLOC` all together if `suites->suiteSz` is 0?  Then adjusting logic inside this function to accomodate.

**Review by cconlon — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 1 above)
Comment (miyazakh) [reply src/ssl.c:12711]: That's true. I think that just skipping the XMALLOC when `suites->suiteSz` is zero is better than the logic here. will check that with CC-RX

**Comment by miyazakh** [comment]
@cconlon 

This is fixed by [PR#6904](https://github.com/wolfSSL/wolfssl/pull/6904)

**Pull request closed without merging**